src/main/java/step3/Car.java

```diff
@@ -1,13 +1,39 @@
 package step3;
 
-public class Car {
+public class Car implements Comparable<Car> {
+  private final String name;
   private int distance;
+  private final CarMovingStartegy carMovingStartegy;
+
+  public Car(CarMovingStartegy carMovingStartegy) {
+    this.name = "";
+    this.carMovingStartegy = carMovingStartegy;
+  }
+
+  public Car(String name, CarMovingStartegy carMovingStartegy) {
+    this.name = name;
+    this.carMovingStartegy = carMovingStartegy;
+  }
+
+  public String getName() {
+    return name;
+  }
 
   public int getDistance() {
     return distance;
   };
 
   public void move() {
-    distance++;
+    if (carMovingStartegy.isCanMove()) {
+      distance++;
+    }
+  }
+
+  @Override
+  public int compareTo(Car target) {
+    if (this.distance == target.distance) {
+      return 0;
+    }
+    return this.distance < target.distance ? -1 : 1;
```

Comment:
`Comparable` 구현 좋습니다! 👍

int 타입의 비교라 아래와 같이 [Integer.compare()](https://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html#compare(int,%20int)) 을 사용할 수 있겠네요.

```suggestion
    return Integer.compare(this.distance, target.distance);
```